typescript/src/deposit-sweep.ts

```diff
@@ -130,238 +150,297 @@ export async function assembleDepositSweepTransaction(
     throw new Error("Number of UTXOs must equal the number of deposit elements")
   }
 
-  const walletKeyRing = createKeyRing(walletPrivateKey, witness)
-  const walletAddress = walletKeyRing.getAddress("string")
+  const network = toBitcoinJsLibNetwork(bitcoinNetwork)
+  const keyPair = ecFactory(tinysecp).fromWIF(walletPrivateKey, network)
+  const walletAddress = addressFromKeyPair(keyPair, network, witness)
 
-  const inputCoins = []
+  // Calculate the value of transaction's output. Note that the value of fee
+  // needs to be subtracted from the sum.
   let totalInputValue = BigNumber.from(0)
-
   if (mainUtxo) {
-    inputCoins.push(
-      bcoin.Coin.fromTX(
-        bcoin.MTX.fromRaw(mainUtxo.transactionHex, "hex"),
-        mainUtxo.outputIndex,
-        -1
-      )
-    )
     totalInputValue = totalInputValue.add(mainUtxo.value)
   }
-
   for (const utxo of utxos) {
-    inputCoins.push(
-      bcoin.Coin.fromTX(
-        bcoin.MTX.fromRaw(utxo.transactionHex, "hex"),
-        utxo.outputIndex,
-        -1
-      )
-    )
     totalInputValue = totalInputValue.add(utxo.value)
   }
+  totalInputValue = totalInputValue.sub(fee)
 
-  const transaction = new bcoin.MTX()
+  // Create the transaction.
+  const transaction = new Transaction()
 
-  transaction.addOutput({
-    script: bcoin.Script.fromAddress(walletAddress),
-    value: totalInputValue.toNumber(),
-  })
+  // Add the transaction's inputs.
+  if (mainUtxo) {
+    transaction.addInput(
+      mainUtxo.transactionHash.reverse().toBuffer(),
+      mainUtxo.outputIndex
+    )
+  }
+  for (const utxo of utxos) {
+    transaction.addInput(
+      utxo.transactionHash.reverse().toBuffer(),
+      utxo.outputIndex
+    )
+  }
 
-  await transaction.fund(inputCoins, {
-    changeAddress: walletAddress,
-    hardFee: fee.toNumber(),
-    subtractFee: true,
-  })
+  // Add transaction output.
+  const scriptPubKey = address.toOutputScript(walletAddress, network)
+  transaction.addOutput(scriptPubKey, totalInputValue.toNumber())
 
-  if (transaction.outputs.length != 1) {
-    throw new Error("Deposit sweep transaction must have only one output")
+  // Sign the main UTXO input if there is main UTXO.
+  if (mainUtxo) {
+    const inputIndex = 0 // Main UTXO is the first input.
+    const previousOutput = Transaction.fromHex(mainUtxo.transactionHex).outs[
+      mainUtxo.outputIndex
+    ]
+
+    await signMainUtxoInput(
+      transaction,
+      inputIndex,
+      previousOutput.script,
+      previousOutput.value,
```

Comment:
Why not pass the whole previous output?

Reply:
Done in 783a3bb9d9328e550f738ed4be9b1ba588859cb9.